### 상관관계 확인

In [2]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity="all"

In [3]:
import numpy as np
import pandas as pd

df = pd.read_csv('./kaggle/input/spaceship-titanic/train_update.csv')
df01 = pd.read_csv('./kaggle/input/spaceship-titanic/train.csv')

In [ ]:
df_died = df[df['Transported']==0]

In [44]:
# # Cabin_port
# df_died.groupby('Cabin_port').count()
# df.groupby('Cabin_port').count()
# # HomePlanet
# df_died.groupby('HomePlanet').count()
# df.groupby('HomePlanet').count()
# # CryoSleep
# df_died.groupby('CryoSleep').count()
# df.groupby('CryoSleep').count()
# # Cabin
df_died.groupby('Cabin').count()
df.groupby('Cabin').count()

len(df_died.groupby('Cabin'))

PassengerId  HomePlanet  CryoSleep  Cabin_num  Cabin_port  Destination  \
Cabin                                                                           
0              129         129        129        129         129          129   
1              207         207        207        207         207          207   
2              239         239        239        239         239          239   
3              282         282        282        282         282          282   
4              630         630        630        630         630          630   
5             1583        1583       1583       1583        1583         1583   
6             1240        1240       1240       1240        1240         1240   
7                5           5          5          5           5            5   

        Age   VIP  RoomService  FoodCourt  ShoppingMall   Spa  VRDeck  Name  \
Cabin                                                                         
0       129   129          129        129           129   129     129   125   
1       207   207          207        207           207   207     207   203   
2       239   239          239        239           239   239     239   234   
3       282   282          282        282           282   282     282   278   
4       630   630          630        630           630   630     630   614   
5      1583  1583         1583       1583          1583  1583    1583  1541   
6      1240  1240         1240       1240          1240  1240    1240  1217   
7         5     5            5          5             5     5       5     4   

       Transported  
Cabin               
0              129  
1              207  
2              239  
3              282  
4              630  
5             1583  
6             1240  
7                5

PassengerId  HomePlanet  CryoSleep  Cabin_num  Cabin_port  Destination  \
Cabin                                                                           
0              256         256        256        256         256          256   
1              779         779        779        779         779          779   
2              747         747        747        747         747          747   
3              495         495        495        495         495          495   
4             1025        1025       1025       1025        1025         1025   
5             2821        2821       2821       2821        2821         2821   
6             2564        2564       2564       2564        2564         2564   
7                6           6          6          6           6            6   

        Age   VIP  RoomService  FoodCourt  ShoppingMall   Spa  VRDeck  Name  \
Cabin                                                                         
0       256   256          256        256           256   256     256   249   
1       779   779          779        779           779   779     779   763   
2       747   747          747        747           747   747     747   733   
3       495   495          495        495           495   495     495   485   
4      1025  1025         1025       1025          1025  1025    1025  1005   
5      2821  2821         2821       2821          2821  2821    2821  2747   
6      2564  2564         2564       2564          2564  2564    2564  2506   
7         6     6            6          6             6     6       6     5   

       Transported  
Cabin               
0              256  
1              779  
2              747  
3              495  
4             1025  
5             2821  
6             2564  
7                6

8

In [54]:
df_died.groupby('Cabin_num').count()['PassengerId'][354]

3

In [1]:
# 비율 확인
def died_persent(column_name):
    
    for i in range(len(df_died.groupby(column_name))):
        try:
            df_d0 = df_died.groupby(column_name).count()['PassengerId'][i] 
            df_0 = df.groupby(column_name).count()['PassengerId'][i]  
            print(f'{column_name} {i} : 전체 {df_0} 중 {df_d0}명 {round(df_d0/df_0*100,2)}% 실종')
        except:
            pass

died_persent('VIP')

NameError: name 'df_died' is not defined

In [95]:
# 비율 저장
import os
import pandas as pd

def died_rate(column_name):
    # 폴더 생성
    try:
        if not os.path.exists('EDAdata'):
            os.makedirs('EDAdata')
    except OSError:
        print ('Error: Creating directory. EDAdata')

    # 파일 경로
    file_path = './EDAdata/' + 'died_rate' + '.csv'

    if os.path.isfile(file_path): # 파일 존재

        entire = column_name + '_entire'
        died = column_name + '_died'
        rate =column_name + '_rate'

        df_rate_new = pd.DataFrame(columns=['index',entire,died,rate])
        
        for index in range(len(df_died.groupby(column_name))):
            try:
                df_d0 = df_died.groupby(column_name).count()['PassengerId'][index] 
                df_0 = df.groupby(column_name).count()['PassengerId'][index]
                df_rate_new = df_rate_new.append(pd.DataFrame([[index,df_0,df_d0,round(df_d0/df_0*100,2)]], columns=['index',entire,died,rate]), ignore_index=True)
            except:
                pass
        
        df_rate = pd.read_csv(file_path)
        df_concat = pd.concat([df_rate,df_rate_new],axis=1)
        df_concat.to_csv(file_path, index=False)

    else: # 파일 존재하지 않음
        entire = column_name + '_entire'
        died = column_name + '_died'
        rate =column_name + '_rate'

        df_rate = pd.DataFrame(columns=['index',entire,died,rate])

        for index in range(len(df_died.groupby(column_name))):
            try:
                df_d0 = df_died.groupby(column_name).count()['PassengerId'][index] 
                df_0 = df.groupby(column_name).count()['PassengerId'][index]
                df_rate = df_rate.append(pd.DataFrame([[index,df_0,df_d0,round(df_d0/df_0*100,2)]], columns=['index',entire,died,rate]), ignore_index=True)
            except:
                pass

        df_rate.to_csv(file_path, index=False)
        

# died_rate('VRDeck')

C:\Users\deepBlue\AppData\Local\Temp\ipykernel_2344\3700331924.py:28: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df_rate_new = df_rate_new.append(pd.DataFrame([[index,df_0,df_d0,round(df_d0/df_0*100,2)]], columns=['index',entire,died,rate]), ignore_index=True)
C:\Users\deepBlue\AppData\Local\Temp\ipykernel_2344\3700331924.py:28: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df_rate_new = df_rate_new.append(pd.DataFrame([[index,df_0,df_d0,round(df_d0/df_0*100,2)]], columns=['index',entire,died,rate]), ignore_index=True)
C:\Users\deepBlue\AppData\Local\Temp\ipykernel_2344\3700331924.py:28: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df_rate_new = df_rate_new.append(pd.DataFrame([[index,df_0,df_d0,round(df_d0/df_0*100,2)]

In [96]:
# file_path = './EDAdata/' + 'died_rate' + '.csv'
# df_rate = pd.read_csv(file_path)


In [74]:
# ## 예시 파일
# column_name = 'Age'
# entire = column_name + '_entire'
# died = column_name + '_died'
# rate =column_name + '_rate'

# df_rate = pd.DataFrame(columns=['index',entire,died,rate])

# for index in range(len(df_died.groupby(column_name))):
#     try:
#         df_d0 = df_died.groupby(column_name).count()['PassengerId'][index] 
#         df_0 = df.groupby(column_name).count()['PassengerId'][index]
#         df_rate = df_rate.append(pd.DataFrame([[index,df_0,df_d0,round(df_d0/df_0*100,2)]], columns=['index',entire,died,rate]), ignore_index=True)
#     except:
#         pass

# # df_rate.head(20)

# # 두 번째 들어갈 때
# column_name = 'RoomService'
# entire = column_name + '_entire'
# died = column_name + '_died'
# rate =column_name + '_rate'

# df_rate_new = pd.DataFrame(columns=['index',entire,died,rate])

# for index in range(len(df_died.groupby(column_name))):
#     try:
#         df_d0 = df_died.groupby(column_name).count()['PassengerId'][index] 
#         df_0 = df.groupby(column_name).count()['PassengerId'][index]
#         df_rate_new = df_rate_new.append(pd.DataFrame([[index,df_0,df_d0,round(df_d0/df_0*100,2)]], columns=['index',entire,died,rate]), ignore_index=True)
#         df_concat = pd.concat([df_rate,df_rate_new],axis=1)
#     except:
#         pass

# df_concat.head(20)

In [75]:
# # 비율 확인

# def died_persent(column_name):
#     df_d0 =df_died.groupby(column_name).count()['PassengerId'][0] #
#     df_d1 = df_died.groupby(column_name).count()['PassengerId'][1] #
#     df_0 = df.groupby(column_name).count()['PassengerId'][0] # 
#     df_1 = df.groupby(column_name).count()['PassengerId'][1] #
#     print(f'{column_name} 0 : 전체 {df_0} 중 {df_d0}명 {round(df_d0/df_0*100,2)}% 실종')
#     print(f'{column_name} 1 : 전체 {df_1} 중 {df_d1}명 {round(df_d1/df_1*100,2)}% 실종')

# died_persent('CryoSleep')

In [29]:
574/3125*100

18.368000000000002

### 도착하지 못한 사람
Transported == 0
1. Cabin_port
    - 좌현(P,0) 전체 4403 중 2405 명 54.6% 실종 
    - 우현(S,0) 전체 4290 중 1910 명 44.5% 실종
2. HomePlanet
    - 비슷한 비율로 실종된것 처럼 보인다 
    - Earth:0 : 전체 4714 중 2719 명 57.7% 실종 
    - Mars:1 : 전체 2175 중 740 명 34.0% 실종 
    - Europa:2 : 전체 1804 중 856 명 47.5% 실종 
3. CryoSleep(방 밖으로 나가지 않는 사람)
    - CryoSleep 0 : 전체 5568 중 3741명 67.19% 실종
    - CryoSleep 1 : 전체 3125 중 574명 18.37% 실종
4. Cabin
    - Cabin 0 : 전체 256 중 129명 50.39% 실종
    - Cabin 1 : 전체 779 중 207명 26.57% 실종
    - Cabin 2 : 전체 747 중 239명 31.99% 실종
    - Cabin 3 : 전체 495 중 282명 56.97% 실종
    - Cabin 4 : 전체 1025 중 630명 61.46% 실종
    - Cabin 5 : 전체 2821 중 1583명 56.11% 실종
    - Cabin 6 : 전체 2564 중 1240명 48.36% 실종
    - Cabin 7 : 전체 6 중 5명 83.33% 실종
    

모든 퍼센트는 소수 2째자리 반올림

In [77]:
df_died.columns

Index(['PassengerId', 'HomePlanet', 'CryoSleep', 'Cabin', 'Cabin_num',
       'Cabin_port', 'Destination', 'Age', 'VIP', 'RoomService', 'FoodCourt',
       'ShoppingMall', 'Spa', 'VRDeck', 'Name', 'Transported'],
      dtype='object')

In [141]:
# Cabin_num 50씩 그룹화
Cabin_num = pd.read_excel('./EDAdata/Cabin_num.xlsx')
Cabin_num = Cabin_num.assign(group="")
# Cabin_num.insert('index_group')
for index in range(len(Cabin_num['index.3'])):
    index_num = Cabin_num['index.3'].iloc[index]
    Cabin_num.loc[index, 'group'] = np.trunc(index_num/10)
Cabin_num
Cabin_num.to_csv('./EDAdata/Cabin_num.csv', index=False)

index.3  Cabin_num_entire  Cabin_num_died  Cabin_num_rate  group
0           0                22              14           63.64    0.0
1           1                15               5           33.33    0.0
2           2                11               6           54.55    0.0
3           3                16               5           31.25    0.0
4           4                 7               5           71.43    0.0
...       ...               ...             ...             ...    ...
1360     1555                 4               3           75.00  155.0
1361     1557                 2               1           50.00  155.0
1362     1559                 1               1          100.00  155.0
1363     1560                 2               2          100.00  156.0
1364     1561                 5               4           80.00  156.0

[1365 rows x 5 columns]

In [142]:
cabin_num = pd.read_csv('./EDAdata/Cabin_num.csv')
df = cabin_num.groupby('group').sum()
df['Cabin_num_rate'] = df['Cabin_num_died']/df['Cabin_num_entire']*100
df = df.reset_index()
df.to_csv('./EDAdata/Cabin_num_group_10.csv', index=False)

### cabin_num 관련 정보
1. 1520 ~ 1529의 
    - 번호를 가진 방에서는 90%가 실종되었다
2. 1530~39, 1560~69, 1510~19의 
    - 번호를 가진 방에서는 80%가 실종되었다
3. 620~19, 1190~99, 1300~09, 470~79, 1450~59, 430~39, 1170~79, 400~09, 1480~89, 1400~09
    - 번호를 가진 방에서는 70%가 실종되었다

- > 관련성 있다는 증거
'CryoSleep','Cabin','Cabin_num','Cabin_port'

외부 내부 관련 정보를

### 저체온증 관련

선장이 식당을 제외한 모든 곳에서 난방과 온수 시스템을 끄라고 지시 -> FoodCourt 와 실종과의 관련성
[해당 링크](https://namu.wiki/w/%ED%83%80%EC%9D%B4%ED%83%80%EB%8B%89%ED%98%B8%20%EC%B9%A8%EB%AA%B0%20%EC%82%AC%EA%B3%A0)